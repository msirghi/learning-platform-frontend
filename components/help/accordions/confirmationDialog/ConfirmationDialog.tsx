import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { useTranslation } from '../../../../i18n';

type Props = {
  open: boolean;
  handleClose: () => void;
  searchInputValue: string;
  onDialogSubmit: () => void;
};

export const QuestionConfirmationDialog: React.FC<Props> = ({
  open,
  handleClose,
  searchInputValue,
  onDialogSubmit
}) => {
  const { t } = useTranslation();

  return (
    <div>
      <Dialog
        maxWidth='lg'
        open={open}
        onClose={handleClose}
        aria-labelledby='alert-dialog-title'
        aria-describedby='alert-dialog-description'
      >
        <DialogTitle id='alert-dialog-title'>{t('help:confirmationDialog.title')}</DialogTitle>
        <DialogContent>
          <DialogContentText id='alert-dialog-description'>
            <div>
              <div>{t('help:confirmationDialog.messageWillBeSentTitle')}:</div>
              <div>
                <b>{searchInputValue}</b>
              </div>
              <div>{t('help:confirmationDialog.confirmYourAction')}</div>
            </div>
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button variant='outlined' onClick={handleClose} color='secondary'>
            {t('help:confirmationDialog.cancelButtonLabel')}
          </Button>
          <Button variant='outlined' onClick={onDialogSubmit} color='primary'>
            {t('help:confirmationDialog.confirmButtonLabel')}
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};
