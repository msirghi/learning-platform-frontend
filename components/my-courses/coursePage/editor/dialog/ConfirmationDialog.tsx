import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import { useTranslation } from '../../../../../i18n';

type Props = {
  message: string;
  onSubmit: () => void;
  onCancel: () => void;
  open: boolean;
};

export const ConfirmationDialog: React.FC<Props> = ({ open, message, onSubmit, onCancel }) => {
  const { t } = useTranslation();

  return (
    <div>
      <Dialog
        open={open}
        onClose={onCancel}
        aria-labelledby='alert-dialog-title'
        aria-describedby='alert-dialog-description'
      >
        <DialogTitle id='alert-dialog-title'>{t('courses:confirmationDialogTitle')}</DialogTitle>
        <DialogContent>
          <DialogContentText id='alert-dialog-description'>{message}</DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={onCancel} color='secondary' variant='outlined'>
            {t('courses:cancel')}
          </Button>
          <Button onClick={onSubmit} color='primary' variant='outlined'>
            {t('courses:continue')}
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};
