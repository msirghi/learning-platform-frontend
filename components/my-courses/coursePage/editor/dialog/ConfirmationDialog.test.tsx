import { Button, Dialog } from '@material-ui/core';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { ConfirmationDialog } from './ConfirmationDialog';

describe('ConfirmationDialog component', () => {
  const defaultProps = {
    message: 'Message',
    onSubmit: jest.fn(),
    onCancel: jest.fn(),
    open: false
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot if all props are provided', () => {
    const wrapper = shallow(<ConfirmationDialog {...defaultProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should set open status with the status from props', () => {
    const wrapper = shallow(<ConfirmationDialog {...defaultProps} open />);
    const dialog = wrapper.find(Dialog);

    expect(dialog).toHaveLength(1);
    expect(dialog.props().open).toBeTruthy();
  });

  it('should fire onSubmit prop function on clicking the submit button', () => {
    const wrapper = shallow(<ConfirmationDialog {...defaultProps} open />);
    const button = wrapper.find(Button).at(1);

    expect(button).toBeDefined();
    button.simulate('click');
    expect(defaultProps.onSubmit).toHaveBeenCalledTimes(1);
  });

  it('should fire onCancel prop function on clicking the cancel button', () => {
    const wrapper = shallow(<ConfirmationDialog {...defaultProps} open />);
    const button = wrapper.find(Button).at(0);

    expect(button).toBeDefined();
    button.simulate('click');
    expect(defaultProps.onCancel).toHaveBeenCalledTimes(1);
  });
});
