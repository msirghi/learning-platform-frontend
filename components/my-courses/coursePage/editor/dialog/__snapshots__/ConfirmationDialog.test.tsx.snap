// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationDialog component should match the snapshot if all props are provided 1`] = `
<div
  data-testid="confirmation-dialog"
>
  <WithStyles(ForwardRef(Dialog))
    aria-describedby="alert-dialog-description"
    aria-labelledby="alert-dialog-title"
    onClose={[MockFunction]}
    open={false}
  >
    <WithStyles(ForwardRef(DialogTitle))
      id="alert-dialog-title"
    >
      courses:confirmationDialogTitle
    </WithStyles(ForwardRef(DialogTitle))>
    <WithStyles(ForwardRef(DialogContent))
      data-testid="confirmation-dialog-content"
    >
      <WithStyles(ForwardRef(DialogContentText))
        id="alert-dialog-description"
      >
        Message
      </WithStyles(ForwardRef(DialogContentText))>
    </WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogActions))>
      <WithStyles(ForwardRef(Button))
        color="secondary"
        onClick={[MockFunction]}
        variant="outlined"
      >
        courses:cancel
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[MockFunction]}
        variant="outlined"
      >
        courses:continue
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(DialogActions))>
  </WithStyles(ForwardRef(Dialog))>
</div>
`;
